src/destructors.md

```diff
@@ -0,0 +1,95 @@
+# Destructors
+
+When an [intialized] [variable] in Rust goes out of scope or a [temporary] is
+no longer needed its _destructor_ is run. [Assignment] also runs the destructor
+of its left-hand operand, unless it's an unitialized variable. If a [struct]
+variable has been partially initialized, only its initialized fields are
+dropped.
+
+The destrutor of a type consists of
+
+1. Calling its [`std::ops::Drop::drop`] method, if it has one.
+2. Recursively running the destructor of all of its fields.
+    * The fields of a [struct], [tuple] or [enum variant] are dropped in
+      declaration order. \*
+    * The elements of an [array] or (owned) [slice][array] are dropped from the
+      first element to the last. \*
+    * The captured values of a [closure] are dropped in an unspecified order.
+    * [Trait objects] run the destructor of the underlying type.
+    * Other types don't result in any further drops.
+
+\* This order was stabilized in [RFC 1857].
+
+Variables are dropped in reverse order of declaration. Variables declared in
```

Comment:
I think this should be a part of `block expressions`, but I'm not going to worry about that for this PR.